openedx/features/course_experience/views/course_outline.py

```diff
@@ -151,3 +154,22 @@ def user_enrolled_after_completion_collection(self, user, course_key):
             return user_enrollment.created > begin_collection_date
         except CourseEnrollment.DoesNotExist:
             return False
+
+    def mark_first_unit_to_resume(self, course_block_tree):
```

Comment:
This is an optional suggestion, you can choose to ignore:
I think this function can be less verbose when written as a recursive one:
```
def mark_first_unit_to_resume(self, block_node):
    children = block.get('children')
    if children:
        children[0]['resume_block' = True
        self.mark_first_unit_to_resume(children[0])
```

Reply:
Oh, yeah that's good. I was trying to clean up the function but everything I tried just looked worse than the original.